beacon-chain/db/kv/state.go

```diff
@@ -294,6 +294,28 @@ func (s *Store) saveStatesEfficientInternal(ctx context.Context, tx *bolt.Tx, bl
 			if err := valIdxBkt.Put(rt[:], validatorKeys[i]); err != nil {
 				return err
 			}
+		case *ethpb.BeaconStateCapella:
```

Comment:
This case switch is a good candidate for simplification using generics since all cases are the same function based on pbState.

Reply:
Yes, this was my impression too. But I didn't want to refactor existing code in this PR.

Reply:
There isn't even a need for generics if we can just pass in fastssz.Marshaler as an interface (beacon state satisfies this)